modules/Emails/include/ListView/ListViewDataEmails.php

```diff
@@ -269,209 +191,817 @@ public function getListViewData(
+
+            if (!empty($where)) {
+                foreach ($filter_fields as $filteredField => $filteredFieldValue) {
+
+                    // Ignore blank fields
+                    if (empty($filteredField)) {
+                        continue;
+                    }
 
-                if($folderType === 'draft' && $emailHeader['draft'] === 0)
-                {
-                    continue;
+                    if (!isset($_REQUEST[$filteredField.'_advanced']) && !isset($_REQUEST[$filteredField.'_basic'])) {
+                        continue;
+                    }
+
+                    if (empty($_REQUEST[$filteredField.'_advanced']) && empty($_REQUEST[$filteredField.'_basic'])) {
+                        continue;
+                    }
+
+                    // strip out the suffix to the the field names
+                    if ((stristr($filteredField, 'advanced') !== false) || (stristr($filteredField, 'basic') !== false)) {
+                        $f = str_ireplace('_advanced', '', $filteredField);
+                        $f = str_ireplace('_basic', '', $f);
+                        if (isset(self::$mapServerFields[$f])) {
+                            $filter[self::$mapServerFields[$f]] = $filteredFieldValue;
+                        }
+
+                        if(array_key_exists($filteredField, $f)) {
+                            continue;
+                        }
+
+                        // if field name is not an imap field
+                        if (!array_key_exists($f, self::$mapServerFields)) {
+                            $searchType = 'crm';
+                        }
+                    } else {
+                        // use the field names
+                        if(in_array($filteredField, self::$mapIgnoreFields)) {
+                            continue;
+                        }
+
+                        // if field name is not an imap field
+                        if (!array_key_exists($filteredField, self::$mapServerFields)) {
+                            $searchType = 'crm';
+                        } else {
+                            if (!empty($_REQUEST[$filteredField.'_advanced'])) {
+                                $filter[self::$mapServerFields[$filteredField]] = $_REQUEST[$filteredField.'_advanced'];
+                            } else if (!empty($_REQUEST[$filteredField.'_basic'])) {
+                                $filter[self::$mapServerFields[$filteredField]] = $_REQUEST[$filteredField.'_basic'];
+                            } else {
+                                $f = str_ireplace('_advanced', '', $filteredField);
+                                $f = str_ireplace('_basic', '', $f);
+                                $filter[self::$mapServerFields[$filteredField]] = $f;
+                            }
+                        }
+                    }
                 }
+            }
+
+            // carry out the filter type
+            switch ($searchType) {
```

Comment:
$searchType can be 'crm' and 'imap' only? handle the default case for accidental wrong or unexpected values